app/src/main/java/org/eyeseetea/malariacare/fragments/AVFragment.java

```diff
@@ -0,0 +1,221 @@
+    }
+
+    private void initDetailedRecyclerViewList() {
+        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(mAVFragment.getActivity().getApplicationContext());
+        mDetailedRecyclerView = (RecyclerView)  getView().findViewById(R.id.detailed_media_list);
+        mDetailedRecyclerView.setHasFixedSize(true);
+        mDetailedRecyclerView.setLayoutManager(linearLayoutManager);
+        mDetailedRecyclerView.setAdapter(detailedAdapter);
+}
+
+    @Override
+    public void onResume() {
+        Log.d(TAG, "onResume");
+        registerFragmentReceiver();
+        super.onResume();
+    }
+
+    /**
+     * Inits adapter.
+     */
+    private void initAdapters() {
+        this.cardViewAdapter = new AVAdapter(this.mMedias);
+        this.detailedAdapter = new AVDetailedAdapter(this.mMedias, mAVFragment.getActivity().getApplicationContext());
+    }
+
+    @Override
+    public void onStop() {
+        Log.d(TAG, "onStop");
+        unregisterFragmentReceiver();
+
+        super.onStop();
+    }
+
+    public void reloadMedia(List<MediaDB> newListMediaDBs) {
+        Log.d(TAG, "reloadMedia (Thread: " + Thread.currentThread().getId() + "): "
+                + newListMediaDBs.size());
+        this.mMedias.clear();
+        this.mMedias.addAll(Media.fromModel(newListMediaDBs));
+        this.cardViewAdapter.notifyDataSetChanged();
+        this.detailedAdapter.notifyDataSetChanged();
+    }
+
+    public void reloadHeader(Activity activity) {
+    }
+
+    /**
+     * Register a survey receiver to load surveys into the listadapter
+     */
+    public void registerFragmentReceiver() {
+        Log.d(TAG, "registerSurveysReceiver");
+
+        if (surveyReceiver == null) {
+            surveyReceiver = new AVFragment.SurveyReceiver();
+            LocalBroadcastManager.getInstance(getActivity()).registerReceiver(surveyReceiver,
+                    new IntentFilter(SurveyService.ALL_MEDIA_ACTION));
+        }
+    }
+
+
+    /**
+     * Unregisters the survey receiver.
+     * It really important to do this, otherwise each receiver will invoke its code.
+     */
+    public void unregisterFragmentReceiver() {
+        Log.d(TAG, "unregisterFragmentReceiver");
+        if (surveyReceiver != null) {
+            LocalBroadcastManager.getInstance(getActivity()).unregisterReceiver(surveyReceiver);
+            surveyReceiver = null;
+        }
+    }
+
+    public void reloadData() {
+        //Reload data using service
+        Intent mediaIntent = new Intent(
+                PreferencesState.getInstance().getContext().getApplicationContext(),
+                SurveyService.class);
+        mediaIntent.putExtra(SurveyService.SERVICE_METHOD, SurveyService.ALL_MEDIA_ACTION);
+        PreferencesState.getInstance().getContext().getApplicationContext().startService(
+                mediaIntent);
+    }
+
+    public void changeList() {
+        toggleLists();
+    }
+
+    private void toggleLists() {
```

Comment:
the toggle button should be always visible when the view mode change between grid and list